quickstarts/microsoft.kubernetes/aks-azure-linux-osguard/main.bicep

```diff
@@ -0,0 +1,67 @@
+@description('The name of the Managed Cluster resource.')
+param clusterName string = 'aks101cluster'
+
+@description('The location of the Managed Cluster resource.')
+param location string = resourceGroup().location
+
+@description('Optional DNS prefix to use with hosted Kubernetes API server FQDN.')
+param dnsPrefix string
+
+@description('Disk size (in GB) to provision for each of the agent pool nodes. This value ranges from 0 to 1023. Specifying 0 will apply the default disk size for that agentVMSize.')
+@minValue(0)
+@maxValue(1023)
+param osDiskSizeGB int = 0
+
+@description('The number of nodes for the cluster.')
+@minValue(1)
+@maxValue(50)
+param agentCount int = 3
+
+@description('The size of the Virtual Machine.')
+param agentVMSize string = 'standard_d2s_v3'
+
+@description('User name for the Linux Virtual Machines.')
+param linuxAdminUsername string
+
+@description('Configure all linux machines with the SSH RSA public key string. Your key should include three parts, for example \'ssh-rsa AAAAB...snip...UcyupgH azureuser@linuxvm\'')
+param sshRSAPublicKey string
+
+resource aks 'Microsoft.ContainerService/managedClusters@2022-05-02-preview' = {
+  name: clusterName
+  location: location
+  identity: {
+    type: 'SystemAssigned'
+  }
+  properties: {
+    dnsPrefix: dnsPrefix
+    agentPoolProfiles: [
+      {
+        name: 'agentpool'
+        osDiskSizeGB: osDiskSizeGB
+        count: agentCount
+        vmSize: agentVMSize
+        osType: 'Linux'
+        mode: 'System'
+        osSKU: 'AzureLinuxOSGuard'
+        osDiskType: 'Managed'
+        enableFIPS: true
+        securityProfile: {
+          enableSecureBoot: true
+          enableVTPM: true
+          }
+      }
+    ]
+    linuxProfile: {
```

Comment:
Lets drop the linuxprofile section from the example and the associated parameters at the top of this template. We can probably also drop the osDiskSizeGB from this example to keep it focused on what needs to be set for OSGuard.